src/android/HockeyApp.java

```diff
@@ -27,14 +31,73 @@
 import java.lang.Thread;
 import java.text.SimpleDateFormat;
 import java.util.Calendar;
+import java.util.ArrayList;
+
+import android.app.Activity;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.view.View;
 
 public class HockeyApp extends CordovaPlugin {
 
+    public static final long XWALK_SCREENSHOT_WAIT_MS = 5000;
+    public static final String XWALK_SCREENSHOT_CAPTURE_MSG = "captureXWalkBitmap";
+    public static final String XWALK_SCREENSHOT_BITMAP_MSG = "onGotXWalkBitmap";
+    
     public static boolean initialized = false;
     public static String appId;
+    public static Object monitor = new Object();
+    public static volatile Bitmap bitmap;
     
     private ConfiguredCrashManagerListener crashListener;
+    
+    // Integration with Crosswalk requires:
+    // - Crosswalk engine version 18 or later
+    // - Latest cordova-plugin-crosswalk-webview plugin     
+    private Bitmap getBitmap() {
+        bitmap = null;
+        boolean isCrosswalk = false;
+        try {
+            Class.forName("org.crosswalk.engine.XWalkWebViewEngine");
+            isCrosswalk = true;
+        } catch (Exception e) {
+        }
 
+        if (isCrosswalk) {
+            long start = System.currentTimeMillis();
+            webView.getPluginManager().postMessage(XWALK_SCREENSHOT_CAPTURE_MSG, this);
+            try {
+                // Can't take more than 5 seconds to get a screenshot
+                long wait = XWALK_SCREENSHOT_WAIT_MS;
+                while (bitmap == null && wait > 0) {
+                    synchronized (monitor) {
+                        monitor.wait(wait);
+                    }
+                    wait = XWALK_SCREENSHOT_WAIT_MS - (System.currentTimeMillis() - start);
+                }
+            } catch (InterruptedException e) {
+            }
```

Comment:
Referring to your last comment about catching interrupts.. If an `InterruptedException` happens, the code jumps out of the `while` loop, so actually the `while` loop wouldn't serve any purpose the way you wrote it. Did you really want to put the `try... catch` inside the `while` loop instead?